src/components/interface/filter/index.jsx

```diff
@@ -1,69 +1,133 @@
 import React from 'react';
-import Button from '@material-ui/core/Button';
-import FilterIcon from '@material-ui/icons/PlaylistAdd';
-import EnhancedMenu from './EnhancedMenu';
+import PropTypes from 'prop-types';
+
+import { withStyles } from '@material-ui/core/styles';
+import Input from '@material-ui/core/Input';
+import LensIcon from '@material-ui/icons/Lens';
+import SearchIcon from '@material-ui/icons/Search';
+
+import CheckBox from '../CheckBox';
+import db from '../../../firebase/firebase';
+import FilterMenu from './FilterMenu';
+import styles from '../../styles';
 
 class Filter extends React.Component {
-  state = {
-    anchorEl: null
-  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      options: {},
+      filtered: {
+        academicLevel: [],
+        awardConsideration: [],
+        competitionLevel: [],
+        danceSize: [],
+        danceStyle: [],
+        school: []
+      }
+    };
+  }
 
-  handleClick = (event) => {
-    this.setState({ anchorEl: event.currentTarget });
-  };
+  componentDidMount() {
+    this.subscribe = db.collection('settings').onSnapshot((querySnapshot) => {
+      const options = {};
+      querySnapshot.forEach((doc) => {
+        options[doc.id] = Object.keys(doc.data());
+      });
+      this.setState({ options });
+    });
+  }
 
-  handleClose = () => {
-    this.setState({ anchorEl: null });
-  };
+  componentWillUnmount() {
+    this.subscribe();
+  }
+
+  handleFilterClearAll = () => this.setState({
+    filtered: {
+      academicLevel: [],
+      awardConsideration: [],
+      competitionLevel: [],
+      danceSize: [],
+      danceStyle: [],
+      school: []
+    }
+  })
+
+  handleFilterChecked = (e) => {
+    const { name, value, checked } = e.target;
+    const { handleFilters } = this.props;
+    const { filtered } = this.state;
 
-  handleMenuClose = () => {
-    this.setState({
-      anchorEl: null
+    this.setState((prevState) => {
+      const category = prevState.filtered[name];
+      const filteredValues = checked ? category.concat(value) : category.filter(v => v !== value);
+      return {
+        filtered: {
+          ...prevState.filtered,
+          [name]: filteredValues
+        }
+      };
     });
+
+    handleFilters(filtered);
   };
 
   render() {
-    const { anchorEl } = this.state;
-    const menuItems = [
+    const { filtered, options } = this.state;
+    const { awardConsideration } = filtered;
+    const { classes } = this.props;
+    const choices = [
       {
-        key: 1,
-        caption: 'Academic Level',
-        options: [{ value: 'test', label: 'test' }, { value: 'test', label: 'test' }]
+        checked: awardConsideration.includes('specialAward'),
+        name: 'awardConsideration',
+        value: 'specialAward',
+        label: (
+          <React.Fragment>
+            {'Special Award Only   '}
```

Comment:
are the extra spaces here required? or can we apply styling instead. I believe this could cause problems with a resizing window.